[CS2113-F15-3] NUSFC 24

hwc0419 opened the pull request:
NUSFC 24 offers football enthusiasts a refreshing break from the complexities of 3D games like FIFA or eFootball. Dive into the core thrill of football with our focus on the art of penalty shooting and goalkeeping, celebrating the pure joy and challenge of the game's decisive moments.

HenryGan138 committed:
Merge pull request #16 from HenryGan138/branch-v1.0UiMain

Branch v1.0 UI main and Resolves #7

Commit:
Add directionIsLegal JUnit Test

directionIsLegal asserts that the direction generated by the Ai class is
a legal direction.

runxinghuan committed:
Merge pull request #21 from runxinghuan/branch-JUnit-Test

Add directionIsLegal JUnit Test

Commit:
viewGoal and goalCheck

ymirmeddeb committed:
Merge branch 'master' into my-branch

IncyBot committed:
Merge branch 'master' into Dev-Info

IncyBot committed:
Merge pull request #3 from IncyBot/Dev-Info

Dev info

ymirmeddeb committed:
Merge branch 'master' into my-branch

ymirmeddeb committed:
Merge pull request #22 from ymirmeddeb/my-branch

viewGoal and goalCheck

JiahaoGan committed:
Add UiTest for v1.0

HenryGan138 committed:
Merge pull request #23 from HenryGan138/branch-v1.0UiMain

Add UiTest for v1.0

Commit:
Enable assertions

Enable assertions so that we can use "assert" in the code.

runxinghuan committed:
Merge pull request #25 from runxinghuan/branch-Enable-Assertion

Enable assertions

Commit:
Solve three checkstyle errors

runxinghuan committed:
Merge pull request #27 from runxinghuan/branch-Fix-Checkstyle

Solve three checkstyle errors

Commit:
Change the expected output for text ui test

runxinghuan committed:
Merge pull request #29 from runxinghuan/branch-Text-Ui-Test

Change the expected output for text ui test

Commit:
Make the Ai class more defensive using assert

runxinghuan committed:
Merge pull request #31 from runxinghuan/branch-More-Defensive

Make the Ai class more defensive using assert

hwc0419 committed:
Add Parser

hwc0419 committed:
Merge pull request #36 from AY2324S2-CS2113-F15-3/branch-Parser

Add Parser

Commit:
no message

ymirmeddeb committed:
Merge pull request #37 from ymirmeddeb/my-branch

minor edits v1.0

Commit:
added added assert and logging to printGoalBeforeShot and printGoalAfterShot

ymirmeddeb committed:
Merge pull request #38 from ymirmeddeb/my-branch

assert and logging

JiahaoGan committed:
Edition to the Duke

HenryGan138 committed:
Merge pull request #40 from HenryGan138/master

Edition to the v1.0b

JiahaoGan committed:
Slight change 2 on v1.0b

JiahaoGan committed:
Revert "Slight change 2 on v1.0b"

This reverts commit 0253006a9cf584764b051c13992d3911180144ee.

hwc0419 committed:
Reformat code

HenryGan138 committed:
Revert "Edition to the v1.0b"

HenryGan138 committed:
Merge pull request #45 from AY2324S2-CS2113-F15-3/revert-40-master

Revert "Edition to the v1.0b"

JiahaoGan committed:
Add Assert into goalcheck

hwc0419 committed:
Merge branch 'branch-Parser'

# Conflicts:
#	src/main/java/seedu/duke/CommandList.java
#	src/main/java/seedu/duke/Duke.java
#	src/main/java/seedu/duke/Formatter.java
#	src/main/java/seedu/duke/exception/BadTokenException.java
#	src/main/java/seedu/duke/exception/ProcessInputException.java
#	src/main/java/seedu/duke/ui/Ui.java

JiahaoGan committed:
Add logging into Ui

hwc0419 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp

# Conflicts:
#	src/main/java/seedu/duke/CommandList.java
#	src/main/java/seedu/duke/Duke.java
#	src/main/java/seedu/duke/ui/Ui.java

JiahaoGan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp into branch-v1.0b-assert

# Conflicts:
#	src/main/java/seedu/duke/CommandList.java
#	src/main/java/seedu/duke/Formatter.java

HenryGan138 committed:
Merge pull request #50 from HenryGan138/branch-v1.0b-assert

Add assert and logging into function

hwc0419 committed:
Package exceptions

hwc0419 committed:
Merge pull request #51 from hwc0419/branch-Parser

Package exceptions

IncyBot committed:
Added Shoot Function

IncyBot committed:
Added Penalty Command

IncyBot committed:
Merge pull request #52 from IncyBot/Dev-Info

Added Shoot and Penalty Function

hwc0419 committed:
Add ParserTest for v1.0

hwc0419 committed:
Merge pull request #53 from hwc0419/branch-Parser

Add ParserTest for v1.0

hwc0419 committed:
Add assert and logging to Parser

hwc0419 committed:
Merge pull request #54 from hwc0419/branch-Parser

Add assert and logging to Parser

hwc0419 committed:
Code quality

hwc0419 committed:
Merge branch 'branch-CodeQuality-v1'

# Conflicts:
#	src/main/java/seedu/duke/CommandList.java
#	src/main/java/seedu/duke/Parser.java
#	src/main/java/seedu/duke/ui/Ui.java
#	src/test/java/seedu/duke/ui/UiTest.java

hwc0419 committed:
Merge pull request #59 from hwc0419/branch-CodeQuality-v1

Branch code quality v1

Commit:
Delete text ui test

As we can use JUnit test now.

JiahaoGan committed:
Week 9 Tutorial work

JiahaoGan committed:
Merge branch 'branch-v1.0b-assert'

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master'

# Conflicts:
#	src/main/java/seedu/duke/Formatter.java

JiahaoGan committed:
Add Duke Class back

Commit:
Remove the command to run text ui test

runxinghuan committed:
Merge pull request #61 from runxinghuan/branch-Delete-Text-ui-test

Delete text ui test

IncyBot committed:
Removed Shot Functionality for Penalty

Commit:
Add real penalty shootout setting

Only make changes with shoot, need change again after penalty is used.
Need to change after save command

JiahaoGan committed:
Design the Shooter Class as the parent Class

JiahaoGan committed:
Beginner, Meidum, ExpertSkill class all implements Shooter Class function

JiahaoGan committed:
ArrayList manage the Shooter Class (for polymorphism implementation like different goal picture)

JiahaoGan committed:
Add upgrade command

JiahaoGan committed:
V2.0 Test1

Commit:
Edit the code based on Hao Wen's comments

Commit:
added penalty saving functionnality

ymirmeddeb committed:
Merge pull request #66 from ymirmeddeb/my-branch

Save Penalty

runxinghuan committed:
Revert "Save Penalty"

runxinghuan committed:
Merge pull request #67 from AY2324S2-CS2113-F15-3/revert-66-my-branch

Revert "Save Penalty"

Commit:
added penalty saving functionnality

IncyBot committed:
Updated as per comments

runxinghuan committed:
Merge pull request #64 from runxinghuan/branch-Shootout-Setting

Add real penalty shootout setting

IncyBot committed:
Updated next line issue

Commit:
Add explanation to one of my feature added

I will add a class diagram later.

IncyBot committed:
Merge branch 'master' into Dev-Info

IncyBot committed:
Merge pull request #63 from IncyBot/Dev-Info

Removed Shoot Functionality for Penalty

IncyBot committed:
Added Difficulty Levels

JiahaoGan committed:
Minor change to improve the code

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master'

JiahaoGan committed:
Change for compiling error

JiahaoGan committed:
Merge branch 'master' into branch-v2.0

# Conflicts:
#	src/main/java/seedu/duke/CommandList.java
#	src/main/java/seedu/duke/Duke.java
#	src/main/java/seedu/duke/Formatter.java
#	src/main/java/seedu/duke/SyntaxAnalyser.java
#	src/main/java/seedu/duke/ui/Ui.java

JiahaoGan committed:
Implement Automatically update player function

JiahaoGan committed:
Improve the code of player class

JiahaoGan committed:
Connect MatchStat with the player class

JiahaoGan committed:
Changes to implement V2.0

JiahaoGan committed:
workflows

HenryGan138 committed:
Merge pull request #72 from HenryGan138/branch-v2.0-t1

Branch v2.0 Add Player Class

Commit:
Add a class diagram for MatchStat class

Commit:
Try to makes the UML diagram visible

Commit:
Try to make the UML diagram visible

Commit:
Another try

Commit:
Try to show the diagram using copy and paste

Commit:
Merge branch 'branch-UG-1' of https://github.com/runxinghuan/tp into branch-UG-1

# Conflicts:
#	docs/DeveloperGuide.md

JiahaoGan committed:
Update to DG

JiahaoGan committed:
Merge branch 'branch-DGv1' into branch-v2.0-t1

HenryGan138 committed:
Merge pull request #75 from HenryGan138/branch-v2.0-t1

Update to DG v1

IncyBot committed:
Update DeveloperGuide.md

IncyBot committed:
Merge branch 'master' of https://github.com/IncyBot/tp

IncyBot committed:
Merge branch 'Dev-Info'

IncyBot committed:
Merge pull request #76 from IncyBot/master

Updated Difficulty and DG

runxinghuan committed:
Merge branch 'master' into branch-UG-1

runxinghuan committed:
Merge pull request #70 from runxinghuan/branch-UG-1

Add explanation to one of my feature added

runxinghuan committed:
Merge branch 'master' into my-branch

runxinghuan committed:
Merge pull request #68 from ymirmeddeb/my-branch

Save Penalty

Commit:
added save class to developer guide

ymirmeddeb committed:
Merge pull request #77 from ymirmeddeb/my-branch

save class to DG

Commit:
Change the author of the MatchStat class

runxinghuan committed:
Merge pull request #78 from runxinghuan/branch-Change-Author

Change the author of the MatchStat class

Commit:
Improve the MatchStat class

Add logic for best-of-five kicks rule, and ensure that only 'yes' or
'no' can be entered after a match ends.

Commit:
Add header comments for MatchStat class

runxinghuan committed:
Merge pull request #79 from runxinghuan/branch-Improve-Shootout-Setting

Improve the MatchStat class

Commit:
Enables player and Ai to take turn to shoot

The save function is not working though.

runxinghuan committed:
Merge pull request #81 from runxinghuan/branch-take-turns

Enables player and Ai to take turn to shoot

Commit:
fix Save class

Commit:
Fix a small bug in updating the score

Commit:
fix save command

Commit:
no message

runxinghuan committed:
Merge pull request #84 from ymirmeddeb/my-branch

fix bug in save command

runxinghuan committed:
Merge pull request #83 from runxinghuan/branch-take-turns

Fix a small bug in updating the score

Commit:
Fix a few bugs

Fix the bug of isPlayerTurn is not updated.
Fix the bug of Ai only generating direction 0.
Fix the bug of treating isGoalSaved as isGoal.
Add showing the score after saving.

runxinghuan committed:
Merge pull request #85 from runxinghuan/branch-v2.0-bug-fix

Fix a few bugs

Commit:
Add introduction and quick start parts of the UG

runxinghuan committed:
Merge pull request #88 from runxinghuan/branch-UG-v2.0

Add introduction and quick start parts of the UG

JiahaoGan committed:
TestCoverage

JiahaoGan committed:
Merge branch 'master' into branch-v2.0

JiahaoGan committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp into branch-v2.0

JiahaoGan committed:
MInor change to pull mistakes on update command

Commit:
Ymir edit UG for save command and delete Save class

Commit:
Update the class diagram for MatchStat

Commit:
Add sequential diagram of MatchStat class to the DG

JiahaoGan committed:
Improve the visual effect of the game

JiahaoGan committed:
Add OOP function to implement shoot function concern power and difficulty

JiahaoGan committed:
Adapt shoot direction from 0-2 to 0-8

JiahaoGan committed:
Rename the function to make it more understandable

JiahaoGan committed:
Change to DG

JiahaoGan committed:
Change to UG

iscyng commented on umlDiagram/player.png:
I am unable to see this diagram in the developers guide, is this just a draft or is it the final version?

iscyng commented on docs/DeveloperGuide.md:
```diff
@@ -4,11 +4,67 @@
+![UML Class Diagram](../umlDiagram/updateSkill.png)
 ## Product scope
 ### Target user profile
 
```
Are you using the user stories? If not can delete it since the one that I see is not related to your app

runxinghuan committed:
Merge pull request #91 from runxinghuan/branch-DG-Sequential-Diagram

Branch dg sequential diagram

Heart-of-N1 commented on docs/DeveloperGuide.md:
```diff
@@ -4,11 +4,67 @@
+
+4. **Feedback to the User**: The system informs the user of the successful skill upgrade through UI messages or other feedback mechanisms, providing immediate recognition of their achievement.
+
+![UML Class Diagram](../umlDiagram/updateSkill.png)
```
Adding associations in the form of solid lines between classes instead of listing them out separately can help visualize the interactions between the classes.

Heart-of-N1 commented on docs/DeveloperGuide.md:
```diff
@@ -4,11 +4,67 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Class Structure
```
In general, good job for clear and precise categorization of the contents inside the developer guide!

Heart-of-N1 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +92,315 @@
+
+An alternative to using an enum could be to use integer constants or string constants to represent the difficulty levels. However, enums provide type safety and better code readability, making them a preferred choice for representing a fixed set of values.
+
+## Class Diagrams
```
Adding sequence diagrams for classes can be beneficial to visualize the interactions between various entities in your system.

Heart-of-N1 commented on docs/DeveloperGuide.md:
```diff
@@ -4,11 +4,67 @@
+
+![UML Class Diagram](../umlDiagram/updateSkill.png)
 ## Product scope
 ### Target user profile
```
Specifying the user profile can help narrow down the scope of the functionality codes.

Heart-of-N1 reviewed (commented):
Generally good job. :)

Mahesh1772 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +92,351 @@
+of multiple players. Here is the class diagram of the class:
+
+
+![MatchStatClassDiagram.png](diagrams%2FMatchStatClassDiagram.png)
```
Perhaps you could consider removing the "C" by adding the "hide" command so that it looks closer to the class diagrams covered in our lectures?

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/87927591/e0121992-270a-407a-ae6d-25163b53c719)

Mahesh1772 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +92,351 @@
+example, `shoot 1` is entered by the player.
+
+
+![MatchStatSequential.png](diagrams%2FMatchStatSequential.png)
```
Sequential diagram looks great and well documented, with the correct notation used!

Mahesh1772 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +92,351 @@
+
+## AI Class 
+
+### Overview
+
+The `Ai` class is responsible for generating a random direction for the computer player (AI) to shoot or save in the game. The difficulty level of the game determines the range of directions the AI can choose from.
+
+### Design
+
+The `Ai` class has the following components:
+
+1. `minDirection` and `maxDirection` static variables to store the minimum and maximum direction values based on the difficulty level.
+2. A constructor that takes a `DifficultyLevel` object as input and initializes the `minDirection` and `maxDirection` based on the difficulty level.
+3. A `getAiDirection()` static method that generates a random direction within the range of `minDirection` and `maxDirection`.
+4. Getter methods `getMinDirection()` and `getMaxDirection()` to retrieve the minimum and maximum direction values, respectively.
+
```
Well detailed explanation of the class, its implementation and design structure

Mahesh1772 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +92,351 @@
+
+An alternative to using an enum could be to use integer constants or string constants to represent the difficulty levels. However, enums provide type safety and better code readability, making them a preferred choice for representing a fixed set of values.
+
+## Class Diagrams
```
Utilizing an image of the class diagram like was used for the MatchStat class diagram at the beginning could help with visualization

Mahesh1772 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +92,351 @@
+
+The `Ai` class depends on the `DifficultyLevel` enum to determine the range of directions for the AI. The `Penalty` class uses the `Ai` class to get the AI's direction and compares it with the user's input direction. The `CommandList` class provides the `goalCheck()` method to determine if a goal is scored based on the user's and AI's directions. The `Formatter` class is responsible for printing  ...
+
+## Future Enhancements
+
+1. **Adaptive AI Difficulty**: Implement an adaptive AI difficulty system where the AI's behavior changes based on the player's performance. This could involve adjusting the range of directions or introducing other challenges for the AI.
+
+2. **Penalty Shootout Mode**: Implement a penalty shootout mode where the game is decided by a series of penalty kicks. This could involve additional rules and scoring mechanisms specific to penalty shootouts.
+
+3. **Custom Difficulty Levels**: Allow users to define custom difficulty levels by specifying the range of directions or other parameters that affect the AI's behavior.
+
+4. **Advanced AI Strategies**: Enhance the AI's decision-making capabilities by implementing more advanced strategies and algorithms. This could involve considering factors such as the player's previous shots, game statistics, or other relevant data.
+
+5. **Multiplayer Mode**: Implement a multiplayer mode where multiple players can participate in the game, either locally or over a network.
```
Good job adding future plans and enhancements along with the existing implemented features!

Mahesh1772 reviewed (commented):
Overall, very well explained and detailed guide! Certain aspects like the class diagrams can be standardized to all use the image format

JiahaoGan committed:
Minor changes on code quality

JiahaoGan committed:
Fix class name for player class

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master'

JiahaoGan committed:
Merge branch 'master' into branch-v2.0c

# Conflicts:
#	src/main/java/seedu/duke/CommandList.java

HenryGan138 committed:
Merge pull request #101 from HenryGan138/branch-v2.0c

Branch v2.0c

T0nyLin commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+| EASY        |
+| MEDIUM      |
+| HARD        |
++-------------+
```
I believe this is an Enum? So may I suggest you to include the <Enumeration> in the title of the box.

T0nyLin reviewed (commented):
Good job on the diagrams, and the guide is well detailed and explained!

T0nyLin commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+- `aiDirectionAdjust(int aiDir)`: Modifies the AI's direction to offer a challenge.
+- `rangeAdjust()`: Modifies the shooting range according to the game's difficulty.
+
+  ![UML Class Diagram](diagrams%2Fplayer.png)
```
May I suggest you to group the private and public methods in each class?

T0nyLin commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+- `shootDirectionFormula(int left, int right, int dir, int power)`: Computes the shooting direction.
+- `aiDirectionAdjust(int aiDir)`: Modifies the AI's direction to offer a challenge.
+- `rangeAdjust()`: Modifies the shooting range according to the game's difficulty.
+
```
<img width="505" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/122875764/a79b1dc9-db9a-4117-9f0d-261541074d86">

Is the highlighted one a method name or constructor? If it is a method, then you should follow the camel case format.

RyanTDL commented on docs/DeveloperGuide.md:
```diff
@@ -4,10 +4,7 @@
-
+## Class Structure
 
 ## Product scope
```
Clearly specifying the product scope could help developers better understand the product. A more comprehensive outline of the user stories can provide context of the product, which allows better understanding of the key features and functionalities of your code

RyanTDL commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+
+Subclasses `BeginnerSkill`, `MediumSkill`, and `ExpertSkill` are derived from the `Player` base class, each tailored to represent different skill levels. These subclasses override specific methods to provide behaviors unique to each skill level, ensuring a dynamic and engaging gameplay experience.
+
+### Attributes
+
+- `name`: String - Identifies the player.
+- `matchCount`: int - Tracks the number of matches played.
+- `power`: int - Represents the player's power level.
+- `skill`: int - Indicates the player's skill level.
+
+### Methods
+
+- `printSelfInfo()`: Displays the player's basic information.
+- `printGoalBeforeShoot()`: Prepares the player for a shooting attempt.
+- `printGoalBeforeSave()`: Prepares the player for a saving attempt.
+- `printGoalAfterShoot(boolean goalScored, int direction)`: Shows the outcome of a shooting attempt.
+- `upgradePower(int level)`: Enhances the player's power based on the specified level.
+- `shootDirectionAdjust(int dir)`: Modifies the shooting direction for the player.
+- `shootDirectionFormula(int left, int right, int dir, int power)`: Computes the shooting direction.
+- `aiDirectionAdjust(int aiDir)`: Modifies the AI's direction to offer a challenge.
+- `rangeAdjust()`: Modifies the shooting range according to the game's difficulty.
```
Clear, detailed explanation of the Player Class. Very comprehensive overview of the attributes and methods

fxe025 replied:
Don't need to put methods like printSkill(), setSkill() in a class diagram. They affects readability.
![Screenshot 2024-04-04 153920](https://github.com/nus-cs2113-AY2324S2/tp/assets/110606445/0d651574-4b6a-4f85-a7ca-a6d6ab153455)

RyanTDL commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+
+### Alternatives Considered
+
+#### Alternative Collections for Player Management
+- **HashSet<Player>**: Evaluated for its uniqueness properties and constant-time performance for add, remove, and contains operations. It was ultimately not selected due to the ordered nature of player management required in our game, which is better served by a List implementation.
+
+#### Polymorphism Implementation Methods
+- **Interface-based Polymorphism**: An alternative approach using interfaces to define common behaviors for all players was considered. While this would have provided a clean separation of concerns, the decision to use class inheritance and method overriding was driven by the shared attributes and methods across player types, favoring a more hierarchical structure.
```
I appreciate the developers outlining the alternatives considered, and the rationale for choosing current implementation instead of alternatives. For example, outlining the rationale of using List instead of HashSet due to the ordered nature of data aids my visualisation of how the data is structured

RyanTDL commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+
+4. **Feedback to the User**: The system informs the user of the successful skill upgrade through UI messages or other feedback mechanisms, providing immediate recognition of their achievement.
+
+![UML Class Diagram](diagrams%2FupdateSkill.png)
```
Well documented sequence diagram. Clear presentation of interactions between classes

runxinghuan committed:
Merge pull request #89 from ymirmeddeb/my-branch

UG for save command and delete Save class

fxe025 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+`updateStat(isScoreGoal)` method returns. However, messages need to be printed out for the player to see that his 
+command finishes executing. Thus, the `printGoalAfterShot(isScoreGoal)` method in the `Formatter` class is called. It 
+prints out all the necessary messages for the user.
+
```
Comprehensive description of MatchStat interaction with other classes. Aligns with the sequence diagram as well.

fxe025 commented on docs/DeveloperGuide.md:
```diff
@@ -36,3 +33,470 @@
+3. **Application of Skill Upgrade**: Eligible players will have their skill attribute adjusted to a higher level. This upgrade might unlock new abilities, enhance stats, or offer other in-game advantages.
+
+4. **Feedback to the User**: The system informs the user of the successful skill upgrade through UI messages or other feedback mechanisms, providing immediate recognition of their achievement.
+
```
Good sequence diagram

fxe025 commented on docs/DeveloperGuide.md:
````diff
@@ -36,3 +33,470 @@
+```
+
+The constructor initializes the `minDirection` and `maxDirection` based on the `DifficultyLevel` parameter. The `getAiDirection()` method generates a random integer within the range of `minDirection` and `maxDirection` using the `Random` class from Java's `util` package.
+
````
Good to include alternative designs in DG.

Commit:
Add coin toss feature

Commit:
Add coin toss feature

Commit:
Add header comments to newly added coin toss methods

runxinghuan committed:
Merge pull request #102 from runxinghuan/branch-Toss-Coin

Add toss coin feature

Commit:
added skill and power upgrading to the saver

ymirmeddeb committed:
Merge pull request #161 from ymirmeddeb/my-branch

skill and power upgrading for saver

Commit:
fix minor bug for save command

Commit:
minor bug in save command

Commit:
Change magic numbers in MatchStat to constants

runxinghuan committed:
Merge pull request #163 from runxinghuan/branch-Fix-Magic-Number-MatchStat

Change magic numbers in MatchStat to constants

Commit:
Change a few DG bugs

Use !include Style.puml in every puml diagram to avoid some style errors

runxinghuan committed:
Merge pull request #164 from runxinghuan/branch-DG-bug

Change a few DG bugs

IncyBot committed:
Added ability to create teams

IncyBot committed:
Added customization and ability to use coins for skills

IncyBot committed:
Updated commands to match with player customizations

JiahaoGan committed:
Improve player skill illustration in UG

HenryGan138 committed:
Merge pull request #166 from HenryGan138/branch-v2.1

Improve player skill illustration in UG

Commit:
Improve on the explanation of best-of-five kicks rule

Commit:
Add unit tests for MatchStat

Commit:
Add unit tests for CoinToss

runxinghuan committed:
Merge pull request #167 from runxinghuan/branch-UG-bug

Improve on the explanation of best-of-five kicks rule

runxinghuan committed:
Merge pull request #168 from runxinghuan/branch-More-Test

Add more test for MatchStat and CoinToss

Commit:
Add PPP draft for Xinghuan

runxinghuan committed:
Merge pull request #169 from runxinghuan/branch-PPP

Add PPP draft for Xinghuan

JiahaoGan committed:
Jiahao Gan PPP draft one

Commit:
fix direction numbers in beginner and medium level saver

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp into my-branch

* 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp:
  Add PPP draft for Xinghuan
  Add unit tests for CoinToss
  Add unit tests for MatchStat
  Improve on the explanation of best-of-five kicks rule
  Improve player skill illustration in UG
  Change a few DG bugs
  Change magic numbers in MatchStat to constants

HenryGan138 committed:
Merge pull request #170 from HenryGan138/branch-v2.1

Jiahao Gan PPP draft one

runxinghuan committed:
Merge pull request #165 from IncyBot/Dev-Info

Added ability to create teams and abiliuty to use coins for customization and skills

Commit:
implement help function

ymirmeddeb committed:
Merge branch 'master' into my-branch

ymirmeddeb committed:
Merge pull request #162 from ymirmeddeb/my-branch

fix minor bug for save command and implement help command

Commit:
remove logging

ymirmeddeb committed:
Merge pull request #171 from ymirmeddeb/my-branch

remove logging

Commit:
UG for shoot command

Commit:
minor aesthetic changes

ymirmeddeb committed:
Create ymirmeddeb.md

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp into my-branch

* 'master' of https://github.com/AY2324S2-CS2113-F15-3/tp:
  Create ymirmeddeb.md

Commit:
PPP draft one

ymirmeddeb committed:
Merge pull request #172 from ymirmeddeb/my-branch

UG for shoot command

Commit:
attempt to fix command bug and UG for all commands

Commit:
Block certain commands at certain stages

Commit:
remove incorrect fix for commands out of phase

ymirmeddeb committed:
Merge pull request #173 from ymirmeddeb/my-branch

attempt to fix command bug and UG for all commands

ymirmeddeb committed:
Merge pull request #176 from runxinghuan/branch-Fix-Invalid-Command

Block certain commands at certain stages

IncyBot committed:
Added JUnit Test for Player

IncyBot committed:
Added PPP for IncyBot

IncyBot committed:
Merge pull request #177 from IncyBot/Dev-Info

Added PPP for Sparsh

Commit:
Make CoinToss more OO

runxinghuan committed:
Merge pull request #178 from runxinghuan/branch-CoinToss-More-OO

Make CoinToss more OO

Commit:
Rename the class name from Duke to NUSFC

JiahaoGan committed:
Change command "upgrade" into "setpower"

JiahaoGan committed:
Change "set" in UG

HenryGan138 committed:
Merge pull request #180 from HenryGan138/branch-v2.1a

Change command "upgrade" into "setpower", change the corresponding UG and formatter

runxinghuan committed:
Merge pull request #179 from runxinghuan/branch-Rename

Rename the class name from Duke to NUSFC

JiahaoGan committed:
Rename from Duke to NUSFC

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master' into branch-v2.1a

JiahaoGan committed:
Update and minor edit change in UG shoot command

JiahaoGan committed:
Fix bugs in PE and refine the implementation of shoot

Commit:
Solve two bugs

1. Sometimes match does not end although it should end based on the rule
2. No coin toss for a new match

JiahaoGan committed:
Improve the code quality

Commit:
Extract methods to make executeCommand() method shorter

runxinghuan committed:
Merge pull request #183 from runxinghuan/branch-MatchStat-Bug

Solve two bugs

runxinghuan committed:
Merge pull request #185 from runxinghuan/branch-UI-Shorter-Method

Extract methods to make executeCommand() method shorter

Commit:
Update the MatchStat DG

HenryGan138 committed:
Merge pull request #182 from HenryGan138/branch-v2.1a

Fix bugs in PE and PR review, update contents in UG

JiahaoGan committed:
Add user stories section in DG

JiahaoGan committed:
Minor edit on code

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master' into branch-v2.1a

HenryGan138 committed:
Merge pull request #186 from HenryGan138/branch-v2.1a

Add DG user stories section and improvement on SD diagram

Commit:
Add DG for CoinToss component

Commit:
Add head and tail in command summary

runxinghuan committed:
Merge pull request #187 from runxinghuan/branch-DG-CoinToss

Branch dg coin toss

Commit:
Add instructions for manual testing in DG

runxinghuan committed:
Merge pull request #188 from runxinghuan/branch-Instruction-Manual-Testing

Add instructions for manual testing in DG

Commit:
Edit PPP for Xinghuan

runxinghuan committed:
Merge pull request #189 from runxinghuan/branch-PPP-Final

Edit PPP for Xinghuan

JiahaoGan committed:
Update on documentation

JiahaoGan committed:
Improve on SD diagram

JiahaoGan committed:
Improve on Coding Quality

HenryGan138 committed:
Merge pull request #190 from HenryGan138/branch-v2.1a

Improvement on coding quality and documentation

JiahaoGan committed:
Minor change in documentation

JiahaoGan committed:
Improve on coding quality

Commit:
Edit UG

ymirmeddeb committed:
Merge pull request #192 from ymirmeddeb/my-branch

Edit UG

Commit:
UG for customization command and fix non-integer value bug

ymirmeddeb committed:
Merge pull request #193 from ymirmeddeb/my-branch

UG for customization command and fix non-integer value bug

HenryGan138 committed:
Merge branch 'master' into branch-v2.1a

HenryGan138 committed:
Merge pull request #191 from HenryGan138/branch-v2.1a

Coding quality improve

JiahaoGan committed:
Add test file

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master' into branch-v2.1a

HenryGan138 committed:
Merge pull request #194 from HenryGan138/branch-v2.1a

Add test file

JiahaoGan committed:
Add user stories

HenryGan138 committed:
Merge pull request #195 from HenryGan138/branch-v2.1a

Add user stories in DG

Commit:
Add introduction to the tp website and edit AboutUs

runxinghuan committed:
Merge pull request #196 from runxinghuan/branch-Final-Final-PPP

Add introduction to the tp website and edit AboutUs

JiahaoGan committed:
About us

JiahaoGan committed:
Merge remote-tracking branch 'tpmaster/master' into branch-v2.1a

# Conflicts:
#	docs/AboutUs.md

Commit:
Solve AboutUs bug

HenryGan138 committed:
Merge pull request #197 from HenryGan138/branch-v2.1a

About us change

Commit:
Try to solve for not able to view PPP

runxinghuan committed:
Merge branch 'master' into branch-Final-Final-PPP

runxinghuan committed:
Merge pull request #198 from runxinghuan/branch-Final-Final-PPP

Branch final final ppp

Commit:
Edit the quick start of UG

runxinghuan committed:
Merge pull request #199 from runxinghuan/branch-Edit-UG

Edit the quick start of UG

JiahaoGan committed:
Update in PPP

HenryGan138 committed:
Merge pull request #200 from HenryGan138/branch-v2.1a

Update in PPP
